tests/TestSuite/Mutant/CollectorTest.php

```diff
@@ -0,0 +1,157 @@
+        $this->assertContains($error, $collector->getErrors());
+    }
+
+    public function testAddKillResultToCollector()
+    {
+        $collector = new Collector();
+
+        $kill = $this->getMutant(false, true, false);
+
+        $collector->collect($kill);
+
+        $this->assertEquals(1, $collector->getTotalCount());
+        $this->assertEquals(0, $collector->getErrorCount());
+        $this->assertEquals(1, $collector->getKilledCount());
+        $this->assertEquals(0, $collector->getTimeoutCount());
+        $this->assertEquals(0, $collector->getEscapeCount());
+        $this->assertEquals(1, $collector->getMeasurableTotal());
+        $this->assertEquals(1, $collector->getVanquishedTotal());
+        $this->assertContains($kill, $collector->getKilled());
+    }
+
+    public function testAddTimeoutResultToCollector()
+    {
+        $collector = new Collector();
+
+        $timeout = $this->getMutant(false, false, true);
+
+        $collector->collect($timeout);
+
+        $this->assertEquals(1, $collector->getTotalCount());
+        $this->assertEquals(0, $collector->getErrorCount());
+        $this->assertEquals(0, $collector->getKilledCount());
+        $this->assertEquals(1, $collector->getTimeoutCount());
+        $this->assertEquals(0, $collector->getEscapeCount());
+        $this->assertEquals(1, $collector->getMeasurableTotal());
+        $this->assertEquals(1, $collector->getVanquishedTotal());
+        $this->assertContains($timeout, $collector->getTimeouts());
+    }
+
+    public function testAddEscapedResultToCollector()
+    {
+        $collector = new Collector();
+
+        $escape = $this->getMutant(false, false, false);
+
+        $collector->collect($escape);
+
+        $this->assertEquals(1, $collector->getTotalCount());
+        $this->assertEquals(0, $collector->getErrorCount());
+        $this->assertEquals(0, $collector->getKilledCount());
+        $this->assertEquals(0, $collector->getTimeoutCount());
+        $this->assertEquals(1, $collector->getEscapeCount());
+        $this->assertEquals(1, $collector->getMeasurableTotal());
+        $this->assertEquals(0, $collector->getVanquishedTotal());
+        $this->assertContains($escape, $collector->getEscaped());
+    }
+
+    public function testToArrayGeneratesAllGroupKeys()
+    {
+        $collector = new Collector();
+
+        $array = $collector->toGroupedMutantArray();
+
+        $this->assertArrayHasKey('escaped', $array);
+        $this->assertArrayHasKey('errored', $array);
+        $this->assertArrayHasKey('timeouts', $array);
+        $this->assertArrayHasKey('killed', $array);
+    }
+
+    public function testToArrayContainsCollectedItemsInCorrectGroups()
+    {
+        $collector = new Collector();
+
+        $error = $this->getMutant(true, false, false);
+        $kill = $this->getMutant(false, true, false);
+        $timeout = $this->getMutant(false, false, true);
+        $escape = $this->getMutant(false, false, false);
```

Comment:
If you leave `Collector` as it is please extract this calls to separate methods:
* createErroredMutant
* createKilledMutant
* createTimeoutMutant
* createEscapeMutant

Those booleans are terrifying :D